Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -490,11 +579,27 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
         attachmentInfo.resize(attachments.size());
         int fdIndex = 0;
         for (size_t i = 0; i < attachments.size(); ++i) {
+#if OS(ANDROID)
+            bool isValid = switchOn(attachments[i],
+                [fdPtr, &fdIndex](const UnixFileDescriptor& fd) {
+                    if (!fd)
+                        return false;
+                    fdPtr[fdIndex++] = fd.value();
+                    return false;
```

Comment:
```
                    if (!!fd) {
                        ASSERT(fdPtr);
                        fdPtr[fdIndex++] = fd.value();
                        // return true???
                    }
                    return false;
```

This adds an assert, to be consistent with [line 598](https://github.com/WebKit/WebKit/pull/49714/files#diff-19e014c216f6bfd1c11216423542cd7dab4e48502122fec35cf7b06cbfb65161R598), and compacts the condition since we seem to be returning `false` regardless. Unless the intention was to return `true`, as `processMessage` seems to hint towards.

That means that valid attachments are only the ones that carry valid buffers, right?

Reply:
Actually, this is a good catch 😱

This should be returning `true` here if the file descriptor is valid:

```suggestion
                    if (!fd)
                        return false;
                    fdPtr[fdIndex++] = fd.value();
                    return true;
```

Valid file descriptors need to be considered valid attachments as well!